miio/tests/test_roidmivacuum_miot.py

```diff
@@ -0,0 +1,188 @@
+from datetime import timedelta
+from unittest import TestCase
+
+import pytest
+
+from miio import RoidmiVacuumMiot
+from miio.roidmivacuum_miot import (
+    ChargingState,
+    FanSpeed,
+    PathMode,
+    RoidmiState,
+    SweepMode,
+    SweepType,
+    WaterLevel,
+)
+from miio.vacuumcontainers import DNDStatus
+
+from .dummies import DummyMiotDevice
+
+_INITIAL_STATE = {
+    "auto_boost": 1,
+    "battery_level": 42,
+    "main_brush_life_level": 85,
+    "side_brushes_life_level": 57,
+    "sensor_dirty_remaning_level": 60,
+    "main_brush_left_minutes": 235,
+    "side_brushes_left_minutes": 187,
+    "sensor_dirty_time_left_minutes": 1096,
+    "charging_state": ChargingState.Charging,
+    "fanspeed_mode": FanSpeed.FullSpeed,
+    "current_audio": "girl_en",
+    "clean_area": 27,
+    "error_code": 0,
+    "state": RoidmiState.Paused.value,
+    "double_clean": 0,
+    "edge_sweep": 0,
+    "filter_left_minutes": 154,
+    "filter_life_level": 66,
+    "forbid_mode": '{"time":[75600,21600,1],"tz":2,"tzs":7200}',
+    "led_switch": 0,
+    "lidar_collision": 1,
+    "mop_present": 1,
+    "mute": 0,
+    "station_key": 0,
+    "station_led": 0,
+    # "station_type": {"siid": 8, "piid": 29}, # uint32
+    # "switch_status": {"siid": 2, "piid": 10},
+    "sweep_mode": SweepMode.Smart,
+    "sweep_type": SweepType.MopAndSweep,
+    "timing": '{"time":[[32400,1,3,0,[1,2,3,4,5],0,[12,10],null],[57600,0,1,2,[1,2,3,4,5,6,0],2,[],null]],"tz":2,"tzs":7200}',
+    "path_mode": PathMode.Normal,
+    "progress": 57,
+    "work_station_freq": 1,
+    # "uid": "12345678",
+    "volume": 4,
+    "water_level": WaterLevel.Mop,
+    "total_clean_time_sec": 321456,
+    "total_clean_areas": 345678,
+    "clean_counts": 987,
+}
+
+
+class DummyRoidmiVacuumMiot(DummyMiotDevice, RoidmiVacuumMiot):
+    def __init__(self, *args, **kwargs):
+        self.state = _INITIAL_STATE
+        super().__init__(*args, **kwargs)
+
+
+@pytest.fixture(scope="function")
+def dummyroidmivacuum(request):
+    request.cls.device = DummyRoidmiVacuumMiot()
+
+
+def assertEnum(a, b):
+    assert a == b
+    assert repr(a) == repr(b)
+
+
+@pytest.mark.usefixtures("dummyroidmivacuum")
+class TestRoidmiVacuum(TestCase):
+    def test_VacuumStatus(self):
+        status = self.device.status()
+        assert status.auto_boost == _INITIAL_STATE["auto_boost"]
+        assert status.battery == _INITIAL_STATE["battery_level"]
+        assertEnum(
+            status.charging_state, ChargingState(_INITIAL_STATE["charging_state"])
+        )
```

Comment:
```suggestion
        assert status.charging_state == ChargingState(_INITIAL_STATE["charging_state"])
```
should be enough for all enum checks. If they are equal, their `repr()`s are also equal.